compiler/bcq-tools/generate_bcq_metadata

```diff
@@ -0,0 +1,214 @@
+        "--input_path",
+        type=str,
+        help="Full filepath of the input file.",
+        required=True)
+    parser.add_argument(
+        "-o",
+        "--output_path",
+        type=str,
+        help="Full filepath of the output file.",
+        required=True)
+    parser.add_argument(
+        "-O",
+        "--output_arrays",
+        type=str,
+        help="Original model output arrays",
+        required=True)
+
+    return parser
+
+
+# This function is copied from
+# https://github.com/tensorflow/tensorflow/blob/r2.3/tensorflow/examples/label_image/label_image.py#L26
+def load_graph(model_file):
+    graph = tf.Graph()
+    graph_def = tf.compat.v1.GraphDef()
+
+    with open(model_file, "rb") as f:
+        graph_def.ParseFromString(f.read())
+    with graph.as_default():
+        tf.import_graph_def(graph_def, name="")
+
+    return graph
+
+
+def generate_metadata_header(original_graph, bcq_version, output_arrays):
+    # Generating metadata starts
+    metadata_values = np.array([ONE_START_MAGICNUM])
+
+    # Append BCQ version
+    metadata_values = np.append(metadata_values, bcq_version)
+
+    # Append original output count
+    output_cnt = output_arrays.count(',') + 1
+    metadata_values = np.append(metadata_values, output_cnt)
+
+    return metadata_values
+
+
+def generate_bcq_metadata_v1(flags):
+    """
+    BCQv1 contains following metadata.
+        - The number of each BCQ information set
+    """
+
+    is_valid = True
+    allowed_info_names = [
+        "bcqinfo_do_w_x", "bcqinfo_alpha", "bcqinfo_packed_binary_code",
+        "bcqinfo_number_of_clusters", "bcqinfo_size_of_clusters",
+        "bcqinfo_qbits_of_clusters", "bcqinfo_dequant_weight"
+    ]
+
+    original_graph = load_graph(flags.input_path)
+    original_graph_def = original_graph.as_graph_def()
+
+    prefix_infonames_dict = {}
+
+    for node in original_graph_def.node:
+        if node.op == "Const" and "/bcqinfo_" in node.name:
+            prefix_index = node.name.index("/bcqinfo_")
+            prefix = node.name[:prefix_index]
+            infoname = node.name[prefix_index + 1:]
+
+            if infoname not in allowed_info_names:
+                is_valid = False
+                break
+
+            if prefix not in prefix_infonames_dict:
+                prefix_infonames_dict[prefix] = set()
+
+            prefix_infonames_dict[prefix].add(infoname)
+
+    # All the number of BCQ information should be same
+    num_of_bcqinfo = -1
+    for key in prefix_infonames_dict:
+        infonames = prefix_infonames_dict[key]
+        if num_of_bcqinfo == -1:
+            num_of_bcqinfo = len(infonames)
+        elif num_of_bcqinfo != len(infonames):
+            is_valid = False
+
+    # The number of BCQv1 information should be 6 or 7
+    if num_of_bcqinfo != 6 and num_of_bcqinfo != 7:
+        is_valid = False
+
+    # If BCQ information is invalid, return original model
+    if is_valid == False:
+        return original_graph_def
```

Comment:
Similar comment. How about printing error message (e.g., "invalid BCQ information") and exiting the script?